src/crush/CrushWrapper.cc

```diff
@@ -2347,7 +2347,12 @@ int CrushWrapper::_choose_type_stack(
       underfull_buckets[j].insert(item);
     }
   }
-  ldout(cct, 20) << __func__ << " underfull_buckets " << underfull_buckets << dendl;
+  ldout(cct, 20) << __func__ << " underfull_buckets " << "[";
+  for (auto p = underfull_buckets.begin(); p != underfull_buckets.end(); ++p) {
```

Comment:
I would normally used ranged for, bit in this case our logic really does depend on the position of the element within the vector rather than just the content of the element. The ways to write that with ranged for are more convoluted than the iterator based version. But, I shall see if Kefu's Forward Declare fix works and if it does do that instead.